pkg/operatorstatus/operator_status.go

```diff
@@ -198,13 +200,133 @@ func NewClusterOperatorStatusCondition(conditionType configv1.ClusterStatusCondi
 	}
 }
 
-func printOperandVersions(versions []configv1.OperandVersion) string {
-	versionsOutput := []string{}
-	for _, operand := range versions {
-		versionsOutput = append(versionsOutput, fmt.Sprintf("%s: %s", operand.Name, operand.Version))
+func unsupportedConditions() []configv1.ClusterOperatorStatusCondition {
+	return []configv1.ClusterOperatorStatusCondition{
+		NewClusterOperatorStatusCondition(configv1.OperatorAvailable, configv1.ConditionTrue, ReasonAsExpected, "Cluster API is not yet implemented on this platform"),
+		NewClusterOperatorStatusCondition(configv1.OperatorProgressing, configv1.ConditionFalse, ReasonAsExpected, ""),
+		NewClusterOperatorStatusCondition(configv1.OperatorDegraded, configv1.ConditionFalse, ReasonAsExpected, ""),
+		NewClusterOperatorStatusCondition(configv1.OperatorUpgradeable, configv1.ConditionTrue, ReasonAsExpected, ""),
+	}
+}
+
+func aggregatedStatusConditions(co *configv1.ClusterOperator, releaseVersion string, currentFeatureGates featuregates.FeatureGate) []configv1.ClusterOperatorStatusCondition {
+	type controllerCondition struct {
+		available configv1.ClusterStatusConditionType
+		degraded  configv1.ClusterStatusConditionType
+	}
+
+	// Define the controller conditions
+	controllerConditions := []controllerCondition{
+		{CapiInstallerControllerAvailableCondition, CapiInstallerControllerDegradedCondition},
+		{CoreClusterControllerAvailableCondition, CoreClusterControllerDegradedCondition},
+		{InfraClusterControllerAvailableCondition, InfraClusterControllerDegradedCondition},
+		{KubeconfigControllerAvailableCondition, KubeconfigControllerDegradedCondition},
+		{SecretSyncControllerAvailableCondition, SecretSyncControllerDegradedCondition},
```

Comment:
Since not all of these have any reason to be degraded, do we require them all to always set a degraded condition?

Would it be better to factor this some way so that we only actually have conditions that are likely to change over time?